src/PageContent/Directory/DirectoryPage.js

```diff
@@ -0,0 +1,256 @@
+import React, { Component } from 'react';
+import DirectoryCard from './DirectoryCard.js';
+import firebase from '../../FirebaseConfig.js';
+import {AccountType, DaysOfWeek, StringFormat} from '../../Enums.js';
+import moment from 'moment';
+import './Directory.css';
+const db = firebase.database();
+
+// import DirectoryFilter from './DirectoryFilter.js'; // TODO: DirectoryFilter Implementation
+
+// @leon 04/02/2018: No logo data found in database, temporarily use this placeholder
+const logoURL = 'https://upload.wikimedia.org/wikipedia/commons/8/89/Portrait_Placeholder.png';
+
+class DirectoryPage extends Component {
+    
+    constructor(props) {
+        super(props);
+        this.state = {
+            umbrellaId: '',
+            raList: [],
+            dgList: [],
+            daList: [],
+        };
+    }
+
+    async componentDidMount() {
+        let umbrellaId = await this.getUmbrellaId();
+        await this.setState({umbrellaId: umbrellaId});
+        this.fetchOrgs();
+    }
+
+    componentWillUnmount() {
+        // detach listeners
+        db.ref('accounts').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+        db.ref('donating_agencies').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+    }
+
+    async getUmbrellaId() {
```

Comment:
we added `umbrella` field to `DONATING_AGENCY_MEMBER` account type a while ago, so you don't need this special switch case anymore

Reply:
Updated!